src/main/java/com/google/gfx/services/matching/MatchingService.java

```diff
@@ -0,0 +1,38 @@
+package com.google.gfx.services.matching;
+
+import com.google.gfx.model.Match;
+import com.google.gfx.model.Order;
+
+/**
+ * Matches existing orders with other existing or new ones.
+ */
+public interface MatchingService {
+
+  /**
+   * This method was conceived to be ran as part of a cron job that matches existing buy and sell
+   * orders.
+   */
+  void matchExistingBuyAndSellOrders();
+
+  /**
+   * Creates a match for an order, that must be accepted by the order owner.
+   *
+   * @param order original posted order
+   * @param amount amount of the match, which might fully or partially match the original order
+   */
+  void matchOrder(Order order, float amount);
```

Comment:
What should this method do, in your opinion?

I was thinking it creates a Match for an order, for a certain amount, since you can have a fractional match.

Why would a collection of matches be returned?